From cf85f054000783d1ed34edd9247f4b9b5b3adea6 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Fri, 9 Sep 2022 17:06:34 +0300
Subject: [PATCH 12/30] s32cc: Update addresses for platform modules

All of these modules will be included to MMU mappings.

Issue: ALB-9287
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/include/s32_clocks.h               | 17 +++++++++--------
 plat/nxp/s32/include/s32_platform_def.h         | 15 ++++++++++++---
 plat/nxp/s32/include/s32_sramc.h                |  5 ++---
 plat/nxp/s32/s32_bl2_el3.c                      |  4 ++++
 plat/nxp/s32/s32_sramc.c                        |  4 +++-
 plat/nxp/s32/s32g/include/s32g_platform_def.h   | 11 +++++++++--
 .../s32/s32g/s32g2/include/s32g2_platform_def.h |  4 ++++
 .../s32/s32g/s32g3/include/s32g3_platform_def.h | 10 ++++++++++
 plat/nxp/s32/s32g/s32g3/s32g3_sramc.c           |  5 +----
 plat/nxp/s32/s32r/include/s32r_platform_def.h   |  7 ++++++-
 10 files changed, 60 insertions(+), 22 deletions(-)

diff --git a/plat/nxp/s32/include/s32_clocks.h b/plat/nxp/s32/include/s32_clocks.h
index 3a625b425..f6aa8e21a 100644
--- a/plat/nxp/s32/include/s32_clocks.h
+++ b/plat/nxp/s32/include/s32_clocks.h
@@ -125,10 +125,13 @@ enum s32g_refclk {
 
 
 #define MC_RGM_BASE_ADDR	0x40078000
-#define MC_ME_BASE_ADDR		0x40088000
-#define RDC_BASE_ADDR		0x40080000
+#define MC_ME_BASE_ADDR		(0x40088000ul)
+#define MC_ME_SIZE		(0xf34)
+#define RDC_BASE_ADDR		(0x40080000ul)
+#define RDC_SIZE		(0x90)
 
-#define S32_FXOSC_BASE_ADDR	0x40050000ul
+#define S32_FXOSC_BASE_ADDR	(0x40050000ul)
+#define S32_FXOSC_SIZE		(0x8)
 
 /*
  * MC_CGM configuration
@@ -141,15 +144,13 @@ enum s32g_mc_cgm {
 };
 
 #define MC_CGM0_BASE_ADDR		(0x40030000ul)
+#define MC_CGM0_SIZE			(0x708)
 #define MC_CGM1_BASE_ADDR		(0x40034000ul)
-#if defined(PLAT_s32r)
-#define MC_CGM2_BASE_ADDR		(0x440C0000ul)
-#else
-#define MC_CGM2_BASE_ADDR		(0x44018000ul)
-#endif
+#define MC_CGM1_SIZE			(0x308)
 #ifndef MC_CGM5_BASE_ADDR
 #define MC_CGM5_BASE_ADDR		(0x40068000ul)
 #endif
+#define MC_CGM5_SIZE			(0x308)
 
 /* This should be kept in sync with other defines in this file,
  * as it cannot be determined at run-time.
diff --git a/plat/nxp/s32/include/s32_platform_def.h b/plat/nxp/s32/include/s32_platform_def.h
index a63802b15..415a5f53e 100644
--- a/plat/nxp/s32/include/s32_platform_def.h
+++ b/plat/nxp/s32/include/s32_platform_def.h
@@ -51,9 +51,16 @@
  */
 #define COUNTER_FREQUENCY	    0x004C4B40
 
-#define SIUL2_0_BASE_ADDR		0x4009C000UL
+#define SIUL2_0_BASE_ADDR		(0x4009C000UL)
+#define SIUL2_0_SIZE			(0x179C)
 
-#define GPR_BASE_ADDR		0x4007C400UL
+#define OCOTP_BASE_ADDR			(0x400A4000UL)
+#define OCOTP_SIZE			(0x400)
+
+/* A53 Cluster GPR */
+#define GPR_BASE_ADDR			(0x4007C400UL)
+#define GPR_BASE_PAGE_ADDR		(0x4007C000UL)
+#define GPR_SIZE			(0xb4)
 #define GPR06_OFF			0x18U
 #define GPR09_OFF			0x24U
 #define GPR36_OFF			0x90U
@@ -77,7 +84,6 @@
 /* GIC (re)definitions */
 #define S32GEN1_GIC_BASE	0x50800000
 #define PLAT_GICD_BASE		S32GEN1_GIC_BASE
-#define S32GEN1_GIC_SIZE	0x100000
 /* SGI to use for kicking the secondary cores out of wfi */
 #define S32_SECONDARY_WAKE_SGI	15
 
@@ -168,6 +174,9 @@
 
 #define S32_FLASH_BASE		(0x0)
 
+#define I2C4_BASE_ADDR		(0x402DC000UL)
+#define I2C4_SIZE		(0x7)
+
 #define USDHC_BASE_ADDR		(0x402f0000ull)
 #define USDHC_SIZE		(0x160)
 
diff --git a/plat/nxp/s32/include/s32_sramc.h b/plat/nxp/s32/include/s32_sramc.h
index cbcb1f3c2..4cc577a95 100644
--- a/plat/nxp/s32/include/s32_sramc.h
+++ b/plat/nxp/s32/include/s32_sramc.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020-2021 NXP
+ * Copyright 2020-2022 NXP
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -10,9 +10,8 @@
 
 #define SRAMC0_BASE_ADDR        0x4019C000
 #define SRAMC1_BASE_ADDR        0x401A0000
-#define SRAMC_SIZE              0x3000
 
-#define SSRAMC_BASE_ADDR        0x44028000
+#define SRAMC_SIZE              0x14
 
 #ifndef __ASSEMBLER__
 #include <stddef.h>
diff --git a/plat/nxp/s32/s32_bl2_el3.c b/plat/nxp/s32/s32_bl2_el3.c
index d14339f3c..5af7b2e13 100644
--- a/plat/nxp/s32/s32_bl2_el3.c
+++ b/plat/nxp/s32/s32_bl2_el3.c
@@ -39,12 +39,16 @@
 
 #define PER_GROUP3_BASE		(0x40300000UL)
 #define FCCU_BASE_ADDR		(PER_GROUP3_BASE + 0x0000C000)
+#define FCCU_SIZE		(0x94)
 #define FCCU_NCF_S1			(FCCU_BASE_ADDR + 0x84)
 #define FCCU_NCFK			(FCCU_BASE_ADDR + 0x90)
 #define FCCU_NCFK_KEY		(0xAB3498FE)
 
 #define MEMORY_STRING		"memory"
 
+#define DDRSS_BASE_ADDR		(0x40380000)
+#define DDRSS_SIZE		(0x80000)
+
 void add_fip_img_to_mem_params_descs(bl_mem_params_node_t *params,
 					    size_t *index)
 {
diff --git a/plat/nxp/s32/s32_sramc.c b/plat/nxp/s32/s32_sramc.c
index 5f14c38b2..fc9ef2a8b 100644
--- a/plat/nxp/s32/s32_sramc.c
+++ b/plat/nxp/s32/s32_sramc.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2020-2021 NXP
+ * Copyright 2020-2022 NXP
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -88,10 +88,12 @@ static void clear_sram_range(struct sram_ctrl *c, uintptr_t start_addr,
 	clear_sramc_range(base, start_offset, end_offset);
 }
 
+#ifdef SSRAMC_BASE_ADDR
 void s32_ssram_clear(void)
 {
 	clear_sramc_range(SSRAMC_BASE_ADDR, 0x0, SSRAM_MAX_ADDR);
 }
+#endif
 
 int s32_sram_clear(uintptr_t start, uintptr_t end)
 {
diff --git a/plat/nxp/s32/s32g/include/s32g_platform_def.h b/plat/nxp/s32/s32g/include/s32g_platform_def.h
index 9b664d4d4..0eff8ae13 100644
--- a/plat/nxp/s32/s32g/include/s32g_platform_def.h
+++ b/plat/nxp/s32/s32g/include/s32g_platform_def.h
@@ -15,7 +15,8 @@
 #include <tbbr_img_def.h>
 #include <s32_platform_def.h>
 
-#define SIUL2_1_BASE_ADDR	0x44010000UL
+#define SIUL2_1_BASE_ADDR	(0x44010000UL)
+#define SIUL2_1_SIZE		(0x17B0)
 #define S32_MAX_I2C_MODULES 5
 
 /* GPR36 */
@@ -24,18 +25,24 @@
 #define CA53_1_2_RVBARADDR_39_32_OFF	(16)
 #define CA53_1_3_RVBARADDR_39_32_OFF	(24)
 
+#define SSRAMC_BASE_ADDR        0x44028000
+
 #define S32G_SSRAM_BASE		(0x24000000)
 #define S32G_SSRAM_SIZE		(0x8000)
 #define S32G_SSRAM_LIMIT	(S32G_SSRAM_BASE + S32G_SSRAM_SIZE)
 
 /* RTC definitions space */
-#define S32G_RTC_BASE		0x40060000
+#define S32G_RTC_BASE		(0x40060000ul)
+#define S32G_RTC_SIZE		(0x18)
 #define RTC_RTCC_OFFSET		0x4
 #define RTC_RTCS_OFFSET		0x8
 #define RTC_RTCS_RTCF		BIT(29)
 #define RTC_APIVAL_OFFSET	0x10
 #define RTC_RTCVAL_OFFSET	0x14
 
+#define WKPU_BASE_ADDR		(0x40090000ul)
+#define WKPU_SIZE		(0x38)
+
 /* Physical address 0x0 is actually mapped; to increase our
  * chances of detecting a 'null pointer access', use a location
  * that is currently not mapped to anything
diff --git a/plat/nxp/s32/s32g/s32g2/include/s32g2_platform_def.h b/plat/nxp/s32/s32g/s32g2/include/s32g2_platform_def.h
index 065f0e5d9..a38bf53d7 100644
--- a/plat/nxp/s32/s32g/s32g2/include/s32g2_platform_def.h
+++ b/plat/nxp/s32/s32g/s32g2/include/s32g2_platform_def.h
@@ -10,6 +10,10 @@
 
 #define PLATFORM_CORE_COUNT		4
 #define PLAT_GICR_BASE			(S32GEN1_GIC_BASE + 0x80000)
+#define S32GEN1_GIC_SIZE		(0x100000)
+
+#define MC_CGM2_BASE_ADDR		(0x44018000ul)
+#define MC_CGM2_SIZE			(0x580)
 
 /* MPIDR_EL1 for the four A53 cores is as follows:
  *	A53_0_cpu0:	0x8000_0000
diff --git a/plat/nxp/s32/s32g/s32g3/include/s32g3_platform_def.h b/plat/nxp/s32/s32g/s32g3/include/s32g3_platform_def.h
index e9a19f890..3d28535be 100644
--- a/plat/nxp/s32/s32g/s32g3/include/s32g3_platform_def.h
+++ b/plat/nxp/s32/s32g/s32g3/include/s32g3_platform_def.h
@@ -10,6 +10,16 @@
 
 #define PLATFORM_CORE_COUNT		8
 #define PLAT_GICR_BASE			(S32GEN1_GIC_BASE + 0x100000)
+#define S32GEN1_GIC_SIZE		(0x200000)
+
+#define MC_CGM2_BASE_ADDR		(0x44018000ul)
+#define MC_CGM2_SIZE			(0x580)
+
+#define MC_CGM6_BASE_ADDR		(0x4053c000ul)
+#define MC_CGM6_SIZE			(0x400)
+
+#define SRAMC2_BASE_ADDR		(0x4055a000ul)
+#define SRAMC3_BASE_ADDR		(0x4055e000ul)
 
 /* MPIDR_EL1 for the eight A53 cores is as follows:
  *	A53_0_cpu0:	0x8000_0000
diff --git a/plat/nxp/s32/s32g/s32g3/s32g3_sramc.c b/plat/nxp/s32/s32g/s32g3/s32g3_sramc.c
index 0920ba862..324213678 100644
--- a/plat/nxp/s32/s32g/s32g3/s32g3_sramc.c
+++ b/plat/nxp/s32/s32g/s32g3/s32g3_sramc.c
@@ -1,14 +1,11 @@
 /*
- * Copyright 2021 NXP
+ * Copyright 2021-2022 NXP
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
 #include <platform_def.h>
 #include <s32_sramc.h>
 
-#define SRAMC2_BASE_ADDR        0x4055A000
-#define SRAMC3_BASE_ADDR        0x4055E000
-
 #define SRAM_BANK_SIZE          (S32_SRAM_SIZE / 4)
 
 #define SRAM_BANK_MIN(N)        (S32_SRAM_BASE + (N) * SRAM_BANK_SIZE)
diff --git a/plat/nxp/s32/s32r/include/s32r_platform_def.h b/plat/nxp/s32/s32r/include/s32r_platform_def.h
index c8aedba2a..a2b9c775b 100644
--- a/plat/nxp/s32/s32r/include/s32r_platform_def.h
+++ b/plat/nxp/s32/s32r/include/s32r_platform_def.h
@@ -10,6 +10,10 @@
 
 #define PLATFORM_CORE_COUNT		4
 #define PLAT_GICR_BASE			(S32GEN1_GIC_BASE + 0x80000)
+#define S32GEN1_GIC_SIZE		(0x100000)
+
+#define MC_CGM2_BASE_ADDR		(0x440C0000ul)
+#define MC_CGM2_SIZE			(0x408)
 
 #define S32_MPIDR_CPU_MASK		0x1
 #define S32_MPIDR_CPU_MASK_BITS	0x1
@@ -49,6 +53,7 @@
 #define S32_MAX_I2C_MODULES 3
 
 /* Off-Chasis */
-#define SIUL2_1_BASE_ADDR	0x4403C000UL
+#define SIUL2_1_BASE_ADDR	(0x4403C000UL)
+#define SIUL2_1_SIZE		(0x17A4)
 
 #endif /* S32_PLATFORM_H */
-- 
2.17.1

